CoreFoundation/CMakeLists.txt

```diff
@@ -450,49 +520,87 @@ target_compile_options(CoreFoundation
                          -Wno-int-conversion
                          -Wno-unused-function
                          -Wno-microsoft-enum-forward-reference)
+target_compile_options(CFURLSessionInterface
+                       PRIVATE
+                         -Wno-shorten-64-to-32
+                         -Wno-deprecated-declarations
+                         -Wno-unreachable-code
+                         -Wno-conditional-uninitialized
+                         -Wno-unused-variable
+                         -Wno-int-conversion
+                         -Wno-unused-function
+                         -Wno-microsoft-enum-forward-reference)
 
 if(NOT CMAKE_SYSTEM_NAME STREQUAL Darwin)
   target_link_libraries(CoreFoundation
                         PRIVATE
-                          ${CURL_LIBRARIES}
                           ${LIBXML2_LIBRARIES})
+  target_link_libraries(CFURLSessionInterface
+                        PRIVATE
+                          ${CURL_LIBRARIES})
 endif()
 if(CMAKE_SYSTEM_NAME STREQUAL Android)
   target_link_libraries(CoreFoundation
                         PRIVATE
                           log)
+  target_link_libraries(CFURLSessionInterface
+                        PRIVATE
+                          log)
 endif()
 target_link_libraries(CoreFoundation
                       PRIVATE
                         Threads::Threads
                         ${CMAKE_DL_LIBS})
+target_link_libraries(CFURLSessionInterface
+                      PRIVATE
+                        Threads::Threads
+                        ${CMAKE_DL_LIBS})
 if(CMAKE_SYSTEM_NAME STREQUAL Windows)
   target_link_libraries(CoreFoundation
                         PRIVATE
                           AdvAPI32
                           Secur32
                           User32
                           mincore)
+  target_link_libraries(CFURLSessionInterface
```

Comment:
I think that you probably don't need to link against most of these libraries.  `WS2_32` and `mincore` are probably all that are needed?

Reply:
I can't test Windows builds at the moment; is it okay if I leave them in and you can remove them later?

Reply:
Sounds fine to me - unlike Unix, the semantics on Windows are that they will be dropped if unused anyways.